lib/shared/gameConfig/game_config.dart

```diff
@@ -0,0 +1,283 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+import 'package:battleship/Weapon/types.dart';
+import 'package:battleship/app_router.dart';
+import 'package:battleship/constants/design/colors.dart';
+import 'package:battleship/constants/design/font.dart';
+import 'package:battleship/game-engine/game.controller.dart';
+import 'package:battleship/game-engine/game.provider.dart';
+import 'package:battleship/game-engine/game.state.dart';
+import 'package:battleship/services/socket.service.dart';
+import 'package:battleship/shared/Button/button_widget.dart';
+import 'package:battleship/shared/Icons/constant.dart';
+import 'package:battleship/shared/Icons/widgets/icon.dart';
+import 'package:battleship/shared/Title/title_widget.dart';
+import 'package:battleship/shared/Weapons/weapons_list.dart';
+
+class GameConfigWidget extends StatefulWidget {
+  const GameConfigWidget({super.key, required this.gameState});
+
+  final GameState gameState;
+
+  @override
+  State<GameConfigWidget> createState() => _GameConfigWidget();
+}
+
+class _GameConfigWidget extends State<GameConfigWidget>
+    with TickerProviderStateMixin {
+  late int currentGridSize;
+  List<Weapon> currentSelectedWeapons = [];
+  int minSize = 8;
+  int maxSize = 12;
+  int defaultBorder = 4;
+  double defaultPadding = 20.0;
+
+  late final TabController _tabController;
+
+  updateUI() {
+    setState(() {});
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _tabController = TabController(length: 2, vsync: this);
+    currentGridSize = widget.gameState.gameSettings!.boardDimensions[0];
+
+    for (var weapon in widget.gameState.gameSettings!.selectedWeapons) {
+      currentSelectedWeapons.add(weapon);
+    }
+  }
+
+  @override
+  void dispose() {
+    _tabController.dispose();
+    super.dispose();
+  }
+
+  void resetGameSettings(GameController gameController) {
+    gameController.updateGridSize(currentGridSize);
+    gameController.resetSelectedWeapons(currentSelectedWeapons);
+    AppRouter().navigateBack(context);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+        padding: const EdgeInsets.all(0.0),
+        child: Column(
+          children: <Widget>[
+            // Title + close button
+            Container(
+              width: double.infinity,
+              padding: EdgeInsets.only(
+                top: defaultPadding / 4,
+                right: defaultPadding,
+                bottom: defaultPadding / 4,
+                left: defaultPadding,
+              ),
+              decoration: BoxDecoration(
+                gradient: LinearGradient(
+                  colors: [BSColors().darkPurple, BSColors().purple],
+                  begin: Alignment.topCenter,
+                  end: Alignment.bottomCenter,
+                ),
+                borderRadius: const BorderRadius.only(
+                    topLeft: Radius.circular(6), topRight: Radius.circular(6)),
+              ),

```

Comment:
Apply the same decoration than the header widget :

```suggestion
              decoration: BoxDecoration(
                border: Border(
                    bottom:
                    BorderSide(color: BSColors().white.withOpacity(0.4), width: 2)),
                gradient: LinearGradient(
                  colors: [BSColors().darkPurple, BSColors().purple],
                  begin: Alignment.topCenter,
                  end: Alignment.bottomCenter,
                ),
              ),

```